.github/workflows/ci.yml

```diff
@@ -37,19 +37,28 @@ jobs:
       - name: Lint
         run: npm run lint
 
-      - name: Test
+      - name: Tests & Coverage
         if: github.ref != 'refs/heads/master'
         run: npm run test
-
-      - name: Test & Coverage
-        if: github.ref == 'refs/heads/master'
-        run: |
-          npm run test
-          export COVERALLS_REPO_TOKEN=${{ secrets.COVERALLS_REPO_TOKEN }}
-          npm run coverage
+        run: npm run coverage
         env:
           CI: true
 
       - name: Benchmark
         if: github.event.inputs.withBenchmark
         run: npm run benchmark
+
+  sonarcloud:
+    name: SonarCloud
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+
```

Comment:
Before mergin this PR, please also run `npm run format`
This should reduce `clean-up` PRs by me and jerome :wink:

Reply:
is that command going to fail if the code is not formatted or is going to format the code?
Maybe we need to run the lint command instead?

Reply:
This is a `yaml` file
we don't have a yaml linter installed
you need to format first and then check with a linter, not the other way around
We could install format git push hooks to force that, but it reduces developer experience while using UI git tools like VSCode :confused:

Reply:
num run lint does not return an error when we run it and code is not formatted then?
Mmmmmm we will find something to help us :)

Reply:
We could try https://github.com/rasshofer/yaml-lint#readme, but I think it doesn't lint errors for blank lines but just indentation errors and so on

So you always need to run format. That's the whole thing why using prettier. It doesn't just format JS/TS for standard syntax but ALL files it can handle! The whole project, everything.

I'm personally against using git commit hooks because they are frustrating and interruptive when using e.g. VS Code's git UI (and I heavily use this)

just run `npm run format && npm run lint && npm run test` before pushing a commit :shrug:  
it's an easy workflow